docs/ota-client-guide/modules/ROOT/pages/bsp-integration.adoc

```diff
@@ -0,0 +1,48 @@
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+
+Adding support for a new board isn't too hard, if you're already familiar with Yocto and you have good documentation about your board's hardware. In this guide, we'll present conceptual information on what steps need to be taken, and then illustrate the concrete steps needed by taking an NXP board (the https://www.nxp.com/design/qoriq-developer-resources/qoriq-ls1043a-development-board:LS1043A-RDB[LS1043A-RDB]) as an example.
+
+All of the changes we're making in this guide will be done in https://github.com/advancedtelematic/meta-updater/[meta-updater]. While you're working on adding support for your new board, you should fork meta-updater. Once the support is ready, please feel free to make a pull request to get it merged into mainline meta-updater.
+
+NOTE: You can also add support for your board by writing a separate layer--see https://github.com/advancedtelematic/meta-updater-raspberrypi[meta-updater-raspberrypi] for an example of this approach.
+
+
+
+== Key Concepts
+
+There are two important concepts about *OSTree* and the *meta-updater* that you need to know before proceeding with the integration.
+
+. *OSTree*
++
+OSTree image deployments are installed inside the `/ostree directory`, and then hard-linked into place in `/boot` at boot time using a bootloader configuration script generated by OSTree when the image is verified.
++
+On an OSTree-managed system, OSTree can tell the bootloader which kernel, initramfs, and device tree blob to load. There are two basic ways to do this:
++
+* on a system with link:https://gitlab.denx.de/u-boot/u-boot/raw/master/doc/README.distro[distroboot], OSTree sets u-boot environment variables to the appropriate values when it switches to a new deployment.
+* In systems that don’t support distroboot (e.g. GRUB) there will be a very minimal, simple bootloader configuration script that is fixed in the bootloader’s partition. The minimal script then points the bootloader to a second, "real" bootloader script, which is managed by OSTree. The way OSTree switches to a new deployment is by replacing the second script so that it points to the new deployment.
++
+NOTE: OSTree will only switch to the new deployment once it has verified that the complete filesystem tree is present; libaktualizr will only instruct OSTree to switch once it has performed xref:uptane.adoc[Uptane] verification on the metadata directing it to switch.
+. *meta-updater*
++
+The https://github.com/advancedtelematic/meta-updater/[meta-updater] contains recipes for the applications and libraries required by libaktualizr and OSTree, along with patches to upstream recipes when modifications are necessary. It also includes board and image classes for the OTA-specific needs of the boards it supports.

```

Comment:
Minor, but it sounds odd to my ear to refer to "the meta-updater". I recommend either just "meta-updater" or "the meta-updater layer".